src/project.coffee

```diff
@@ -323,8 +323,22 @@ class Project extends Model
         # allow ENOENT errors to create an editor for paths that dont exist
         throw error unless error.code is 'ENOENT'
 
-    @bufferForPath(filePath).then (buffer) =>
-      @buildEditorForBuffer(buffer, options)
+    absoluteFilePath = @resolvePath(filePath)
+
+    fileSize = fs.getSizeSync(absoluteFilePath)
+
+    if fileSize >= 20 * 1048576 # 20MB
+      choice = atom.confirm
+        message: 'Atom can have issues with files over 20MB.'
```

Comment:
Oh wait, these editor features are disabled when the file is larger than 2MB, not 20MB. Not sure why I thought these limits were the same.

Would showing that message for files larger than 2MB be something we'd consider? I kind of like the idea of informing users that some features of the editor were disabled. Since a 2MB file limit would be much lower, we might consider adding a config setting to disable this warning so that users who often open large files don't need to click the button every single time?

Just through out some ideas. :bulb: